src/main/java/net/minecraftforge/common/BiomeManager.java

```diff
@@ -123,6 +166,83 @@ public static void removeBiome(BiomeType type, BiomeEntry entry)
 
+    /**
+     * Gets the ocean biome entries for a type of ocean.
+     * This method should be called to get the ocean biome entries for a type of ocean.
+     * This method is threadsafe.
+     *
+     * @param type The type of ocean to get the entries for.
+     * @return An ImmutableList of the ocean biome entries for the ocean type or null if there are no entries.
+     */
+    @Nullable
+    public static ImmutableList<OceanBiomeEntry> getOceanBiomes(OceanType type)
+    {
+        int idx = type.ordinal();
+        synchronized (BiomeManager.class)
+        {
+            List<OceanBiomeEntry> list = idx >= oceanBiomes.length ? null : oceanBiomes[idx];
+            return list != null ? ImmutableList.copyOf(list) : null;
+        }
+    }
+
+    /**
+     * Gets the ocean biome entries that have a biome as their main biome ('main' meaning the biome that is not its deep or mix variant).
+     * This method should be called to get the ocean biome entries that have a biome as their main biome.
+     * This method is threadsafe.
+     *
+     * @param biome The biome to check for.
+     * @param type A specific OceanType to also check for or null for any OceanType.
+     * @param excludeWarm True to exclude WARM OceanType (used internally for mixing oceans)
+     * @return An ImmutableList of ocean biome entries that meet the conditions specified in the params or null if empty
+     */
+    @Nullable
+    public static ImmutableList<OceanBiomeEntry> getOceanBiomesContainingBiome(Biome biome, @Nullable OceanType type, boolean excludeWarm)
+    {
+        OceanType[] values = type != null ? new OceanType[] {type} : OceanType.values();
+        List<OceanBiomeEntry> list = new ArrayList<>();
+        synchronized (BiomeManager.class)
+        {
+            for(OceanType types : values)
+            {
+                for(OceanBiomeEntry entry : getOceanBiomes(types))
+                {
+                    if(entry.biome == biome && (excludeWarm ? type != OceanType.WARM : true))
+                    {
+                        list.add(entry);
+                    }
+                }
+            }
+            return list.isEmpty() ? null : ImmutableList.copyOf(list);
+        }
+    }
+
+    /**
+     * Checks if a biome is present in the ocean biome entries for every OceanType.
+     * This method should be called to check if a biome is present in the ocean biome entries for every OceanType (i.e. if a biome is considered an ocean).
+     * This method is threadsafe.
+     *
+     * @param biome The biome to check
+     * @param shallow If the biome is shallow (i.e. not a deep ocean)
+     * @return True if the biome is an ocean or is a shallow ocean if shallow is true
+     */
+    public static boolean isOceanBiome(Biome biome, boolean shallow)
+    {
+        synchronized (BiomeManager.class)
+        {
+            for(OceanType type : OceanType.values())
+            {
+                for(OceanBiomeEntry entry : getOceanBiomes(type))
```

Comment:
getOceanBiomes can be null.
Also, it might be worth caching this result? Do we expect a lot of biomes in here?

Reply:
First thing I've already answered above.

Caching it would be good if a lot of biomes get added, although it's just cycling through a copy of the ocean biomes for a type, so it shouldn't be that performance impacting.

If you still think caching it is a good idea I can do that 👍